packages/react-components/src/components/datatree-output-component.tsx

```diff
@@ -188,4 +190,51 @@ export class DataTreeOutputComponent extends AbstractOutputComponent<AbstractOut
             this._debouncedFetchSelectionData();
         }
     }
+
+    private async exportOutput(): Promise<void> {
+        const focusContainer = document.getElementById(this.props.traceId + this.props.outputDescriptor.id + 'focusContainer');
+        if (focusContainer) {
+            const table = focusContainer.querySelector('div:nth-child(2) > table');
+            if (table) {
+                const rows = table.querySelectorAll('tr');
+
+                const csvArray = [];
+                for (let i = 0; i < rows.length; i++) {
+                    const row = [];
+                    const cols = rows[i].querySelectorAll('td, th');
+                    for (let j = 0; j < cols.length - 1; j++) {
+                        let data;
+                        const content = cols[j].textContent;
+                        if (content) {
+                            data = content.replace(/\s*\|/g, '');
+                        } else {
+                            data = content;
+                        }
+                        row.push(data);
+                    }
+                    csvArray.push(row.join(','));
+                }
+                const tableString = csvArray.join('\n');
+
+                const link = document.createElement('a');
+                link.setAttribute('href', `data:text/csv;charset=utf-8,${encodeURIComponent(tableString)}`);
+                link.setAttribute('download', this.props.traceName ?? 'export');
```

Comment:
Might be useful to differentiate the outputs in the file name, e.g. 'traceName - outputName.csv'